exporter/kafkaexporter/kafka_exporter.go

```diff
@@ -41,7 +40,12 @@ func (ke kafkaErrors) Error() string {
 }
 
 func (e *kafkaTracesProducer) tracesPusher(_ context.Context, td ptrace.Traces) error {
-	messages, err := e.marshaler.Marshal(td, e.topic)
+	topic := e.cfg.Topic
+	if td.ResourceSpans().Len() > 0 {
+		topic = e.cfg.getTopic(td.ResourceSpans())
+	}
```

Comment:
```suggestion
	topic = e.cfg.getTopic(td.ResourceSpans())
```
Any reason we can't just call the method and have it handle edge cases, and set default value when necessary? (Same comment for logs and metrics)

Reply:
I have implemented it in that way.